tests/fv/auto_ip.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
+set -e
+set -x
+show_commands() {
+dist/calicoctl status
+dist/calicoctl shownodes --detailed
+dist/calicoctl ipv4 pool show
+dist/calicoctl profile show --detailed
+}
+
+# Set it up
+docker rm -f node1 node2 etcd || true
+docker run -d --net=host --name etcd quay.io/coreos/etcd:v2.0.10
+dist/calicoctl reset || true
+
+show_commands
+dist/calicoctl node --ip=127.0.0.1
+dist/calicoctl profile add TEST_GROUP
+
+# Add endpoints
+export DOCKER_HOST=localhost:2377
+while ! docker ps; do
+echo "Waiting for powerstrip to come up"
+  sleep 1
+done
+docker run -e CALICO_IP=auto -tid --name=node1 busybox
+docker run -e CALICO_IP=auto -tid --name=node2 busybox
+
+# Perform a docker inspect to extract the configured IP addresses.
+node1_ip="$(docker inspect node1 | grep IPAddress)"
+node1_ip="${node1_ip#*: \"}"
+node1_ip="${node1_ip%\"*}"
+echo "Node 1 IP address is $node1_ip"
+
+node2_ip="$(docker inspect node2 | grep IPAddress)"
+node2_ip="${node2_ip#*: \"}"
+node2_ip="${node2_ip%\"*}"
+echo "Node 2 IP address is $node2_ip"
+
```

Comment:
I don't think the test beyond here is useful - it's exactly the same as mainline.sh, so just get rid.

You _could_ use a mix of manually assigned addresses instead of just auto, then grep for the known IP and check the return code. But maybe just add a comment to that effect and we'll do the proper checks in the python rewrite.